dev/image-guidelines.md

```diff
@@ -0,0 +1,335 @@
+
+Here is how our fonts allocations compare to the default ones:
+
+![](image-guidelines/excalidraw_default_fonts.png)
+
+We host the font files on our documentation site so that our SVG images can load them.
+
+```
+https://docs.velociraptor.app/fonts/playpen-sans-v13-latin-500.woff2
+```
+
+```
+https://docs.velociraptor.app/fonts/playpen-sans-v13-latin-800.woff2
+```
+
+```
+https://docs.velociraptor.app/fonts/SVBasicManual.woff2
+```
+
+The version of Excalidraw mentioned above can load it's fonts from anywhere, but it's convenient that we can also load the fonts from the above-mentioned URIs. That way you can be sure that you're seeing the same fonts in Excalidraw that will be displayed in the SVG images.
+
+![](image-guidelines/excalidraw_custom_fonts.png)
+
+Unfortunately when the Excalidraw diagrams are exported to SVG the custom fonts are not included. But we can switch the fonts by doing a text search & replace on the SVG files. This is explained [here](#fonts-in-svg-images)
+
+For normal text use the **Medium** font size. It's thickness should correspond closely to the thickness of lines and arrows set to **Extra Bold**.
+
+## Colors
+
+For **screenshot annotations** we use HTML `#228be6` for lines and text.
+This is one of the colors in Excalidraw's built-in palette and it contrasts well with Velociraptor's light theme.
+It's also light enough to be visible against dark backgrounds.
+
+For pure **line diagrams** we use black for lines and text, with de-emphasized elements being grey.
+
+Additional colors can be used to emphasize particular drawing objects.
+
+The default Excalidraw color palette is well designed, so only use those colors.
+This makes it easier to ensure that colors used in future can be matched those used in the past.
+
+## Excalidraw object library
+
+This file is an Excalidraw library containing reusable shape objects for our drawings.
+
+[library.excalidrawlib](library.excalidrawlib)
+
+If you find or create objects that you think will be useful in future drawings then please add them to this library, save the changes and commit the updated version back to the repo.
+
+
+## Screenshots
+
+### Screen size
+
+While we may do our daily work on screens with high display resolutions,
+the documentation will likely be read on smaller displays or possibly even printed.
+It needs to be legible on typical screens without a magnifying glass or zooming in.
+
+Ideally we want screenshots in the documentation to have a comfortably low resolution.
+This can be achieved by either:
+
+- only showing the part of the screen that the user needs to see.
+- using a constrained screen size for full-screen screenshots.
+
+These are explained below.
+
+#### Partial screens
+
+Screenshots that show only part of the screen area are relatively straightforward.
+
+![](image-guidelines/screen_partial.png)
+
+A useful trick in Chrome Dev Tools is that you can right-click on a DOM element and capture an image of just that element.
```

Comment:
I tried this technique but it is less than satisfactory . The first issue is that chrome saves the file instead of copy to the clipboard meaning i have to double click it open in another program then copy paste into excalidraw increasing the time and complexity.

Then I really want to show something like 
![image](https://github.com/user-attachments/assets/9562b4d1-98ec-4575-ac57-621a9b30ca45)

But there is no html node that directly corresponds to - the closest thing is the top pane
![image](https://github.com/user-attachments/assets/7990a0e1-6bbe-4f41-adc2-accceefb2c70)

but you can see the margins are very tight so its not a good screenshot. 

If I need to crop it anyway there is no point in using chrome to take the screenshot.

Reply:
I think we should standardise on the framing of the screenshot though - usually we should use the full headers and sidebar because this gives navigation context as to where the user is in the app. So I do think it is important to think about this.

Reply:
Have you tried «Take screenshot» in Firefox, in the right-click menu? It lets you choose a document element and snap the borders to that, but you may still override it later by, say extending it. It doesn't come with drawing tools, however. It saves both to file and the clipboard.

Reply:
> I think we should standardise on the framing of the screenshot though - usually we should use the full headers and sidebar because this gives navigation context as to where the user is in the app. So I do think it is important to think about this.

Yes, when screenshotting only a part of the screen then it's important to anchor it by including some familiar element like part of the header or sidebar. Unless a previous screenshot has already provided that spacial/navigational context.

Reply:
> I tried this technique but it is less than satisfactory . The first issue is that chrome saves the file instead of copy to the clipboard meaning i have to double click it open in another program then copy paste into excalidraw increasing the time and complexity.
> 
> Then I really want to show something like ![image](https://private-user-images.githubusercontent.com/3856546/368837642-9562b4d1-98ec-4575-ac57-621a9b30ca45.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3MzMyNTUsIm5iZiI6MTcyNjczMjk1NSwicGF0aCI6Ii8zODU2NTQ2LzM2ODgzNzY0Mi05NTYyYjRkMS05OGVjLTQ1NzUtYWM1Ny02MjFhOWIzMGNhNDUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTlUMDgwMjM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDVlZTA0NGQ4MTI2ODJlNWQ3YjQ0NWIxZjg3OWRlY2UwY2RmZTQ3YWY4YjRmNzhiMzZmOTU4ZGJhMTI0N2VmNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.nID4lvXP9LwIwHx-caYbcGobH3yfVJM8qOju5AmTIxs)
> 
> But there is no html node that directly corresponds to - the closest thing is the top pane ![image](https://private-user-images.githubusercontent.com/3856546/368837729-7990a0e1-6bbe-4f41-adc2-accceefb2c70.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3MzMyNTUsIm5iZiI6MTcyNjczMjk1NSwicGF0aCI6Ii8zODU2NTQ2LzM2ODgzNzcyOS03OTkwYTBlMS02YmJlLTRmNDEtYWRjMi1hY2NjZWVmYjJjNzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTlUMDgwMjM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmQ5ZDlhYzAwY2I1MjM2YmI1ZTE4NjAzZDFmODMwOTdmNmVmMDE3ZGE2YWU4ZGExN2UxMTg2MmIyZTIwNjFjMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.7dtcbghS-p-7Y_XhC_Na7064B3SAwtRHRGLzQU6Z51A)
> 
> but you can see the margins are very tight so its not a good screenshot.
> 
> If I need to crop it anyway there is no point in using chrome to take the screenshot.

All valid points. It won't work in all situations but in many situations it's more precise than capturing a screen area by manually doing a rectangle-select with some external tool. With a rectangle-select it's easy to accidentally include edges of adjacent elements or even things that border your virtual screen. That's why it's provided as a tip/trick. I'll annotate it as such.

The tight margins are actually a good thing because Excalidraw is going to add margins.